packages/flutter_tools/lib/src/android/adb.dart

```diff
@@ -204,7 +204,7 @@ class AdbDevice {
   /// Device model; can be null. `XT1045`, `Nexus_7`
   String get modelID => _info['model'];
 
-  set modelID(String value) {
+  void set modelID(String value) {
```

Comment:
Ooh, good news! It already exists: avoid_return_types_on_setters

Hopefully these two lints don't conflict. I'll try it tomorrow.